### ISSUE TITLE
Unable to hide or omit DublinCore fields using form directives on a custom edit form

### ISSUE THREAD

**Issue by bdbaddog** [plone/plone.app.dexterity#30]
_Original author: jimba...@gmail.com (March 30, 2010 02:42:37)_

<b>What steps will reproduce the problem?</b>
1.Attempt to hide or omit DublinCore fields in custom edit form using form
directives. None of the following work:
form.mode(**{'IDublinCore.language': 'hidden'})
form.mode(**{'plone.app.dexterity.behaviors.metadata.IDublinCore.language':
'hidden'})
form.mode(language='hidden')
form.omitted('IDublinCore.language')
form.omitted('plone.app.dexterity.behaviors.metadata.IDublinCore.language')

<b>What is the expected output? What do you see instead?</b>
Expect the field to not appear on the form. It is still there.

<b>What version of the product are you using? On what operating system?</b>
http://good-py.appspot.com/release/dexterity/1.0-next?plone=3.3.5

<b>Please provide any additional information below.</b>

_Original issue: http://code.google.com/p/dexterity/issues/detail?id=114_

**Comment by mschmidt42** [comment]
To always hide the field:

Edit my.package.\_\_init\_\_.py

```python
from plone.app.dexterity.behaviors.metadata import ICategorization
from plone.autoform.interfaces import OMITTED_KEY
from zope.interface import Interface

ICategorization.setTaggedValue(OMITTED_KEY,
                          [(Interface, 'language', 'true')])
```
Save
Restart plone

**Comment by iham** [comment]
@mschmidt42 
yes. ALWAYS. what about: only when my.behavior is applied to a certain type?

**Comment by mschmidt42** [comment]
In this example I didn't find a way to do that. So I hide the field ICategorization everywhere. Then I copied the behavior from plone.app.dexterity.behaviors.metadata, made my own behavior out of it and add it (for all my.type.xml) at my.policy/profiles/default/types/ to the section <property name="behaviors">.

**Comment by mschmidt42** [comment]
I didn't find a way to do that.

**Comment by jensens** [comment]
Closing as outdated. Unable to hide DC fields on Plone 3.3/4. Works correctly in modern Plone.
If you think this issue is still relevant please reopen it. Thank you for the original report!
